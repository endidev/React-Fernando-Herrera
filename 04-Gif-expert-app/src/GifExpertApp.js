import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = []}) => {


    // const [categories, setCategories] = useState(['One Punch'])
    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     // setCategories( [...categories, 'HunterXHunter'])
    //     setCategories( cats => [...cats, 'HxH'])
    // }

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            {/* <button onClick={ handleAdd }>Agregar serie</button> */}

            { 
                categories.map( category => 
                    <GifGrid 
                        key={category}
                        category={category}/>    
                )
            }
        </>
    );
}




export default GifExpertApp;