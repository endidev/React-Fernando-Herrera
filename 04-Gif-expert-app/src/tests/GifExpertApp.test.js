import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp";

describe('Test en GifExpertApp', () => {
    
    
    test('deberia matchear con el snapshot', () => {

        const wrapper = shallow(<GifExpertApp />)

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        const categories = ['One Peace', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
        
    })
    
    
})