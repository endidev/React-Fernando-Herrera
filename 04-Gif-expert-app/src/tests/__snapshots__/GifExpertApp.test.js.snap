// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Peace"
    key="One Peace"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;

exports[`Test en GifExpertApp deberia matchear con el snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
